cmake/templates/HPXMacros.cmake.in

```diff
@@ -93,11 +93,26 @@ function(hpx_setup_allocator)
     string(TOUPPER "${HPX_WITH_MALLOC}" HPX_MALLOC_UPPER)
     string(TOUPPER "${HPX_WITH_MALLOC_DEFAULT}" HPX_MALLOC_DEFAULT_UPPER)
     if(NOT (HPX_MALLOC_UPPER STREQUAL HPX_MALLOC_DEFAULT_UPPER))
-      hpx_error("The specified allocators do not match. This Project is configured with '${HPX_WITH_MALLOC}' and HPX was configured with '${HPX_WITH_MALLOC_DEFAULT}'.")
+      hpx_error("The specified allocators do not match. This Project is "
+        "configured with '${HPX_WITH_MALLOC}' and HPX was configured with "
+        "'${HPX_WITH_MALLOC_DEFAULT}'.")
     endif()
   endif()
   set(HPX_WITH_MALLOC ${HPX_WITH_MALLOC_DEFAULT} CACHE STRING
-    "Define which allocator should be linked in. Options are: system, tcmalloc, jemalloc, tbbmalloc, and custom (default is: tcmalloc)"
-    STRINGS "system;tcmalloc;jemalloc;mimalloc;tbbmalloc;custom")
+    "Define which allocator should be linked in. Options are: system, tcmalloc, "
+    "jemalloc, tbbmalloc, and custom (default is: tcmalloc)")
+  set_property(CACHE HPX_WITH_MALLOC
+    PROPERTY STRINGS "system;tcmalloc;jemalloc;mimalloc;tbbmalloc;custom")
   include(HPX_SetupAllocator)
 endfunction()
+
+# Check we are using the same build type as HPX
+function(hpx_check_cmake_build_type)
+  get_property(_GENERATOR_IS_MULTI_CONFIG GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+  if (NOT _GENERATOR_IS_MULTI_CONFIG AND
+      NOT "${HPX_BUILD_TYPE}" STREQUAL "${CMAKE_BUILD_TYPE}")
```

Comment:
I don't know if you feel like a slightly more complex check is needed, not allowing to mix debug with other types (instead of an exact match).

I report it here, just in case you'd want to use it.

```cmake
  # They may not match just if none of them is a Debug build
  if (  NOT HPX_BUILD_TYPE STREQUAL CMAKE_BUILD_TYPE
          AND
        (HPX_BUILD_TYPE STREQUAL "Debug" OR CMAKE_BUILD_TYPE STREQUAL "Debug")  )
```

Reply:
@albestro yeah, I think expanding the check to do that makes sense. Thanks!

Reply:
LGTM